File: chap12_heart/reentry_loop_map.m
%  -------------------------------------------------------------------
%
%   Plot reentry loop maps.
%
%   For Chapter 12, Section 12.5.3 of
%   Keener and Sneyd, Mathematical Physiology, 3rd Edition, Springer.
% 
%   Written by James Keener and James Sneyd.
% 
%  ------------------------------------------------------------------- 

clear all
close all
clc

set(0,                           ...
   'defaultaxesfontsize', 20,   ...
   'defaultaxeslinewidth', 1.5, ...
   'defaultlinelinewidth', 2.0)

L=1;
Tr = 3/4;
t = [0:.01:5];
dt = [Tr:0.01:10];
c = 1/2+2*(dt-Tr)./(1+(dt-Tr));
Lbyc = L./c;

ndx = min(find(Lbyc<=Tr));

figure(1)
    T=1.75;
    plot(dt(1:ndx),Lbyc(1:ndx),dt(ndx:end),Lbyc(ndx:end),':',t,t,'--',t,Tr*ones(length(t),1),':','linewidth',3)
    hold on
    plot([0,dt(ndx)],[T,T], ':',[dt(ndx),5],[T,T],'linewidth',3)
    axis([0 2.5 0 2.5])
    xlabel('\Delta T_n')
    ylabel('\Delta T_{n+1}')
    text(0.2,0.9,'T_r','fontsize',20)
    text(0.2,1.85,'T','fontsize',20)
    text(0.8,2.1,'L/c','fontsize',20)
    hold off


figure(2)
    T=1.25;
    plot(dt(1:ndx),Lbyc(1:ndx),dt(ndx:end),Lbyc(ndx:end),':',t,t,'--',t,Tr*ones(length(t),1),':','linewidth',3)
    hold on
    plot([0,dt(ndx)],[T,T], ':',[dt(ndx),5],[T,T],'linewidth',3)
    axis([0 2.5 0 2.5])
    xlabel('\Delta T_n')
    ylabel('\Delta T_{n+1}')
    text(0.2,0.9,'T_r','fontsize',20)
    text(0.2,1.35,'T','fontsize',20)
    text(0.8,2.1,'L/c','fontsize',20)

hold off

File: chap12_heart/reentry_loop_map.py

#   -------------------------------------------------------------------
# 
#    Plot reentry loop maps.
# 
#    For Chapter 12, Section 12.5.3 of
#    Keener and Sneyd, Mathematical Physiology, 3rd Edition, Springer.
#  
#    Written by James Keener and James Sneyd.
#  
#   ------------------------------------------------------------------- 

import numpy as np
import matplotlib.pyplot as plt

# Set parameters
L = 1
Tr = 3/4
t = np.arange(0, 5, 0.01)
dt = np.arange(Tr, 10, 0.01)
c = 1/2 + 2 * (dt - Tr) / (1 + (dt - Tr))
Lbyc = L / c

ndx = np.min(np.where(Lbyc <= Tr))

# Plot figure 1
plt.figure(1)
T = 1.75
plt.plot(dt[:ndx], Lbyc[:ndx], dt[ndx:], Lbyc[ndx:], ':', t, t, '--', t, Tr * np.ones(len(t)), ':', linewidth=3)
plt.plot([0, dt[ndx]], [T, T], ':', [dt[ndx], 5], [T, T], linewidth=3)
plt.axis([0, 2.5, 0, 2.5])
plt.xlabel(r'$\Delta T_n$')
plt.ylabel(r'$\Delta T_{n+1}$')
plt.text(0.2, 0.9, r'$T_r$', fontsize=20)
plt.text(0.2, 1.85, r'$T$', fontsize=20)
plt.text(0.8, 2.1, r'$L/c$', fontsize=20)

# Plot figure 2
plt.figure(2)
T = 1.25
plt.plot(dt[:ndx], Lbyc[:ndx], dt[ndx:], Lbyc[ndx:], ':', t, t, '--', t, Tr * np.ones(len(t)), ':', linewidth=3)
plt.plot([0, dt[ndx]], [T, T], ':', [dt[ndx], 5], [T, T], linewidth=3)
plt.axis([0, 2.5, 0, 2.5])
plt.xlabel(r'$\Delta T_n$')
plt.ylabel(r'$\Delta T_{n+1}$')
plt.text(0.2, 0.9, r'$T_r$', fontsize=20)
plt.text(0.2, 1.35, r'$T$', fontsize=20)
plt.text(0.8, 2.1, r'$L/c$', fontsize=20)

plt.show()
